src/stackWalker.cpp

```diff
@@ -333,8 +338,19 @@ int StackWalker::walkVM(void* ucontext, ASGCT_CallFrame* frames, int max_depth,
                             sp = frame.senderSP();
                             fp = *(uintptr_t*)fp;
                         } else {
-                            pc = stripPointer(*(void**)sp);
-                            sp = frame.senderSP();
+                            if (X64) {
+                                char* ptr = (char*)pc;
+                                if (*ptr == 0x41 && *(ptr + 1)== 0x55) { // push %r13
+                                    sp = frame.senderSP();
+                                } else {
+                                    sp = ((uintptr_t*)fp)[InterpreterFrame::sender_sp_offset];
+                                }
+                                pc = stripPointer(((void**)fp)[FRAME_PC_SLOT]);
+                                fp = *(uintptr_t*)fp;
+                            } else {
+                                pc = stripPointer(*(void**)sp);
+                                sp = frame.senderSP();
+                            }
```

Comment:
How about removing the flag `X64`?
```suggestion
#ifdef __x86_64__
                                char* ptr = (char*)pc;
                                if (*ptr == 0x41 && *(ptr + 1)== 0x55) { // push %r13
                                    sp = frame.senderSP();
                                } else {
                                    sp = ((uintptr_t*)fp)[InterpreterFrame::sender_sp_offset];
                                }
                                pc = stripPointer(((void**)fp)[FRAME_PC_SLOT]);
                                fp = *(uintptr_t*)fp;
#else
                                pc = stripPointer(*(void**)sp);
                                sp = frame.senderSP();
#endif
```

Reply:
Thanks, pushed